cr.js

```diff
@@ -19,9 +20,19 @@ function PriorityQueue(size) {
 
     // returns the oldest-added value with the highest priority
     this.Pop = function() {
-        maxKey = Math.max(Object.keys(this.store));
-        this.count--;
-        return this.store[maxKey].shift();
+
+        keys = Object.keys(this.store);
+        if (keys.length > 0) {
+            maxKey = Math.max.apply(Math, keys);
+            this.count--;
+            val =  this.store[maxKey].shift();
+            if (this.store[maxKey].length == 0) {
+                delete this.store[maxKey];
+            }
+            return val;
+        } else {
+            return null;
+        }
```

Comment:
If there is nothing left to pop i.e. this.store = {}, this.store[maxKey] would complain. Also might make sense, to remove the priority key, if there aren't any values corresponding to that key left in this.store